### ISSUE TITLE
Loss_MIT wrong?

### ISSUE THREAD

**Issue by lcnnnn** [WenjieDu/SAITS#32]
I saw that the loss of MIT computaion in core.py was
'MIT_loss = self.customized_loss_func(
                X_tilde_3, inputs["X_ori"], inputs["indicating_mask"] 
            )'
,which computed the MAE between M~3 and X_ori and differed to the paper.

**Comment by WenjieDu** [comment]
The imputation values for X are from X_tilde_3, hence it is fine.

**Comment by lcnnnn** [comment]
Thank  you for your reply , I know that the imputation values for X are from X_tilde_3, just as the code shows:
# combine X_tilde_1 and X_tilde_2
        X_tilde_3 = (1 - combining_weights) * X_tilde_2 + combining_weights * X_tilde_1
        # replace non-missing part with original data
        X_c = masks * X + (1 - masks) * X_tilde_3
but  the equation in paper is L_MIT=MAE(X_c,X,I)

**Comment by WenjieDu** [comment]
Yes, but as I said, the code logic here is right. The parts for loss calculation in `L_MIT=MAE(X_c,X,I)` and 
`MIT_loss = self.customized_loss_func(X_tilde_3, inputs["X_ori"], inputs["indicating_mask"])` both determined by `indicating_mask`, i.e., the term `I`, and they are the same. You can try to replace X_tilde_3 with X_c, the MIT_loss result will not change.

**Comment by lcnnnn** [comment]
Thank&nbsp;you&nbsp;for&nbsp;your&nbsp;sincerely&nbsp;reply!&nbsp;I&nbsp;will&nbsp;try&nbsp;it&nbsp;later

宁成林
***@***.***








---Original---
From: &quot;Wenjie ***@***.***&gt;
Date: Fri, Jan 19, 2024 18:00 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [WenjieDu/SAITS] Loss_MIT wrong? (Issue #32)




 
Yes, but as I said, the code logic here is right. The parts for loss calculation in L_MIT=MAE(X_c,X,I) and
 MIT_loss = self.customized_loss_func(X_tilde_3, inputs["X_ori"], inputs["indicating_mask"]) both determined by indicating_mask, i.e., the term I, and they are the same. You can try to replace X_tilde_3 with X_c, the MIT_loss result will not change.
 
&mdash;

**Comment by WenjieDu** [comment]
My pleasure. If you're working on time series imputation, our recent survey work https://github.com/WenjieDu/Awesome_Imputation may be helpful to you. The benchmark code has been open source, and the preprint paper will be released soon.

**Comment by lcnnnn** [comment]
好的！非常感谢！

宁成林
***@***.***








---Original---
From: &quot;Wenjie ***@***.***&gt;
Date: Fri, Jan 19, 2024 19:01 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [WenjieDu/SAITS] Loss_MIT wrong? (Issue #32)




 
My pleasure. If you're working on time series imputation, our recent survey work https://github.com/WenjieDu/Awesome_Imputation may be helpful to you. The benchmark code has been open source, and the preprint paper will be released soon.
 
&mdash;

**Comment by WenjieDu** [comment]
I am closing this issue because the proposed questions have been solved. If you have further concerns, reopening it will be welcome. 

BTW, you can also join [our PyPOTS community](https://github.com/WenjieDu/PyPOTS/?tab=readme-ov-file#-community) to discuss research related to partially-observed time series data.
